fs/osfs.py

```diff
@@ -266,13 +266,11 @@ def _gettarget(self, sys_path):
         if hasattr(os, "readlink"):
             try:
                 if _WINDOWS_PLATFORM:  # pragma: no cover
-                    target = os.readlink(sys_path)
+                    return os.readlink(sys_path)
                 else:
-                    target = os.readlink(fsencode(sys_path))
+                    return fsdecode(os.readlink(fsencode(sys_path)))
```

Comment:
The link target was returned as `bytes`. A genuine bug uncovered by Mypy. Thanks, Mypy.